Src/Newtonsoft.Json/Linq/JObject.Async.cs

```diff
@@ -41,16 +42,54 @@ public partial class JObject
         /// <param name="cancellationToken">The token to monitor for cancellation requests.</param>
         /// <param name="converters">A collection of <see cref="JsonConverter"/> which will be used when writing the token.</param>
         /// <returns>A <see cref="Task"/> that represents the asynchronous write operation.</returns>
-        public override async Task WriteToAsync(JsonWriter writer, CancellationToken cancellationToken, params JsonConverter[] converters)
+        public override Task WriteToAsync(JsonWriter writer, CancellationToken cancellationToken, params JsonConverter[] converters)
         {
-            await writer.WriteStartObjectAsync(cancellationToken).ConfigureAwait(false);
+            var t = writer.WriteStartObjectAsync(cancellationToken);
+            if (!t.IsCompletedSucessfully())
+            {
+                return AwaitStartObject(t, writer, cancellationToken, converters);
+            }
 
             for (int i = 0; i < _properties.Count; i++)
             {
-                await _properties[i].WriteToAsync(writer, cancellationToken, converters).ConfigureAwait(false);
+                t = _properties[i].WriteToAsync(writer, cancellationToken, converters);
+                if (!t.IsCompletedSucessfully())
+                {
+                    return AwaitProperties(t, i + 1, writer, cancellationToken, converters);
+                }
+            }
+
+            t = writer.WriteEndObjectAsync(cancellationToken);
+
+            return (t.IsCompletedSucessfully()) ? AsyncUtils.CompletedTask : AwaitEndObject(t);
```

Comment:
Is this needed? Couldn't you just return the Task from `writer.WriteEndObjectAsync(cancellationToken);`?

Reply:
Yeah... is what you get for over thinking :)

Reply:
Or underthinking... :/